tests/scripts/all.sh

```diff
@@ -769,6 +782,95 @@ component_build_default_make_gcc_and_cxx () {
     make TEST_CPP=1
 }
 
+component_test_submodule_cmake () {
+    # USE_CRYPTO_SUBMODULE: check that the build works with CMake
+    msg "build: cmake, full config + USE_CRYPTO_SUBMODULE, gcc+debug"
+    scripts/config.pl full # enables md4 and submodule doesn't enable md4
+    scripts/config.pl unset MBEDTLS_MEMORY_BACKTRACE # too slow for tests
+    CC=gcc cmake -D USE_CRYPTO_SUBMODULE=1 -D CMAKE_BUILD_TYPE=Debug .
+    make
+    msg "test: top-level libmbedcrypto wasn't built (USE_CRYPTO_SUBMODULE, cmake)"
+    if_build_succeeded not test -f library/libmbedcrypto.a
+    msg "test: libmbedcrypto symbols are from crypto files (USE_CRYPTO_SUBMODULE, cmake)"
+    if_build_succeeded objdump -g crypto/library/libmbedcrypto.a | grep -E 'crypto/library$' > /dev/null
```

Comment:
If the build failed, `if_build_succeeded objdump -g crypto/library/libmbedcrypto.a` does nothing, so `grep` fails, which defeats the purpose of `if_build_succeeded`, which is to not exit the script immediately in `--keep-going` mode. Workaround:
```suggestion
    if_build_succeeded objdump -g crypto/library/libmbedcrypto.a | if_build_succeeded grep -E 'crypto/library$' > /dev/null
```
But since this idiom is used many times, make a function:
```sh
objdump_must_contain () {
    objdump -g "$1" | grep -E "$2"
}
```
and call `if_build_succeeded objdump_must_contain crypto/library/libmbedcrypto.a 'crypto/library$'`.

Reply:
Also pre-existing, hence IMO out of scope for this PR.

Reply:
This can break CI runs, so I think this is a high-priority fix and a blocker to make this branch the main development branch.

Reply:
It doesn't break CI runs that would have passed, it only breaks --keep-going when the build fails. But you have a point. I'm still inclined to handle that in a follow-up PR (with high priority) because I think reviews are easier if we avoid mixing merging and bugfixes.

Reply:
Resolved here: https://github.com/ARMmbed/mbedtls/pull/2438